docs/upgrading.md

```diff
@@ -1,66 +1,104 @@
-# Upgrading Contour to 0.15
+# Upgrading Contour
 
-## Changes in 0.15
-
-### Breaking change
-The major breaking change in Contour 0.15 is that `contour serve` (the mode that actually serves the control plane for Envoy) now requires that either TLS certificates be available, or you supply the `--insecure` parameter.
-
-**Without one of these options, Contour will not start.**
-
-### Other changes
+This document describes how to upgrade from Contour 0.14 to 0.15.
 
-With the increased security available with supplying TLS certificates for securing gRPC, we are recommending that users migrate to separate Contour and Envoy pods, deployed using a Deployment and Daemonset respectively. This will soon enable greater functionality for status updates on Contour-managed objects, and should help with upgrade issues eventually.
-
-The other major feature that is available is an experimental version of Leader Election. Currently, you can enable leader election with the `--enable-leader-election` flag to `contour serve`. This will ensure that *only the elected leader serves the gRPC endpoints.* That is, only the elected leader will actually open the gRPC port and start serving.
-
-There are some further requirements for this at the moment, see the howto section below for more.
+Please read this upgrade guide carefully, as there is a required
+change for this upgrade. Not performing this change *will leave you
+with Contour not running*.
 
 ## The easy way to upgrade
 
 If the following are true for you:
 
- * Your installation is in the `heptio-contour` namespace
- * You are using one of the current example deployments
- * Your workload can take a minute or two of downtime
+ * Your installation is in the `heptio-contour` namespace.
+ * You are using one of the example deployments.
+ * Your workload can take a minute or two of downtime.
+
+Then you can upgrade to 0.15 by deleting the `heptio-contour` namespace and reinstalling
```

Comment:
Then the simplest way to upgrade to 0.15  is to delete the `heptio-contour` namespace and reapply one of the example configurations.
From rom the root directory of the repository: